merkle/compact/tree.go

```diff
@@ -43,21 +43,21 @@ func (r RootHashMismatchError) Error() string {
 // TODO(pavelkalinnikov): Remove it, use compact.Range instead.
 type Tree struct {
 	hasher hashers.LogHasher
+	rng    *Range
 	root   []byte
-	// the list of "dangling" left-hand nodes, NOTE: index 0 is the leaf, not the root.
-	nodes [][]byte
-	size  int64
-}
-
-func isPerfectTree(x int64) bool {
-	return x != 0 && (x&(x-1) == 0)
 }
 
 // GetNodeFunc is a function prototype which can look up particular nodes
 // within a non-compact Merkle tree. Used by the compact Tree to populate
 // itself with correct state when starting up with a non-empty tree.
 type GetNodeFunc func(depth int, index int64) ([]byte, error)
 
+// nodeID identifies a node of a Merkle tree.
+type nodeID struct {
```

Comment:
This is going to create confusion with storage.NodeID. It's more like the coordinates of a node so could be renamed.

Reply:
Is it? It's not exported, so the scope is quite limited, seems fine to me...

Reply:
I still think renaming it would be clearer.